Mathlib/Algebra/Polynomial/Bivariate.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2024 Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu
+-/
+import Mathlib.Algebra.Polynomial.AlgebraMap
+
+/-!
+# Bivariate polynomials
+
+This file introduces the notation `R[X][Y]` for the polynomial ring `R[X][X]` in two variables,
+and the notation `Y` for the second variable, in the `PolynomialPolynomial` scope.
```

Comment:
```suggestion
and the notation `Y` for the second variable, in the `Polynomial2` scope.
```
would be better maybe? It would match the existing `Sum3` scope which plays a similar role for `α ⊕ β ⊕ γ`.

Reply:
I agree it's a good idea to rename (pending approval of @Multramate), though your analogy is not very exact (`Sum` only appears twice in `α ⊕ β ⊕ γ` so it could be appropriate to call it `Sum2` or `SumSum`).

Reply:
I don't mind changing it to `Polynomial2` since `Sum3` exists. I would think it's unlikely that someone will create `R[X][Y][Z]` - I observed many months ago that working in this ring was really slow, but maybe that is a lot better now.

Reply:
Ah, I just saw your other comment - now I think that we should definitely change `evalEval` to `eval2` if we want to change `PolynomialPolynomial` to `Polynomial2`....

Reply:
Don't you find `eval2` too confusing given [Polynomial.eval₂](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Polynomial/Eval.html#Polynomial.eval%E2%82%82)? Should we change `Polynomial2` back then @YaelDillies? Or should we make a poll on Zulip ...

Reply:
I guess we could also just put the notations under the Polynomial scope ...